call.go

```diff
@@ -213,10 +213,19 @@ func invoke(ctx context.Context, method string, args, reply interface{}, cc *Cli
 			// RPC has completed or failed.
 			put func()
 		)
+		codec := c.codec
+		if codec == nil {
+			codec = cc.dopts.codec
+		}
+		contentSubtype := c.contentSubtype
+		if contentSubtype == "" {
+			contentSubtype = codec.String()
+		}
 		// TODO(zhaoq): Need a formal spec of fail-fast.
 		callHdr := &transport.CallHdr{
-			Host:   cc.authority,
-			Method: method,
+			Host:           cc.authority,
+			Method:         method,
+			ContentSubtype: contentSubtype,
```

Comment:
See comments on wire protocol - right now, grpc-go effectively is always saying that `proto` is the default subtype.